org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -337,76 +324,82 @@ private void runAnalysisAndUpdateMarkers(final IProgressMonitor monitor, IProjec
     return usedConfigurators;
   }
 
-  private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource, AnalysisResults result) throws CoreException {
-    ITextFileBufferManager iTextFileBufferManager = FileBuffers.getTextFileBufferManager();
-    if (iTextFileBufferManager == null) {
+  private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource, AnalysisResults result, TriggerType triggerType) throws CoreException {
+    ITextFileBufferManager textFileBufferManager = FileBuffers.getTextFileBufferManager();
+    if (textFileBufferManager == null) {
       return;
     }
+
     Set<IFile> failedFiles = result.failedAnalysisFiles().stream().map(f -> f.<IFile>getClientObject()).collect(Collectors.toSet());
-    PreviousMarkerCache markerCache = new PreviousMarkerCache(this.request, failedFiles);
-    long now = new Date().getTime();
     for (Entry<IResource, List<Issue>> resourceEntry : issuesPerResource.entrySet()) {
-      IResource r = resourceEntry.getKey();
-      if (failedFiles.contains(r)) {
+      IResource resource = resourceEntry.getKey();
+      if (failedFiles.contains(resource)) {
         continue;
       }
-      List<IMarker> previousMarkers = markerCache.getPrevious(r);
-      Object lastAnalysis = r.getSessionProperty(LAST_ANALYSIS_PROP_NAME);
-      Long creationTimeStampForNewIssues;
-      if (previousMarkers.isEmpty() && lastAnalysis == null) {
-        creationTimeStampForNewIssues = null;
+      if (resource instanceof IFile) {
+        trackIssues(resource, resourceEntry.getValue(), triggerType);
       } else {
-        creationTimeStampForNewIssues = now;
-      }
-      List<Issue> rawIssues = resourceEntry.getValue();
-      try {
-        if (r instanceof IFile) {
-          issueTrackingOnFile(iTextFileBufferManager, r, previousMarkers, rawIssues, creationTimeStampForNewIssues);
-        } else {
-          issueTracking(r, previousMarkers, rawIssues, null, creationTimeStampForNewIssues);
-        }
-      } catch (Exception e) {
-        SonarLintCorePlugin.getDefault().error("Unable to compute position of SonarLint marker on resource " + r.getName(), e);
+        // TODO handle non-file-level issues
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)